circuits/src/transaction.rs

```diff
@@ -57,7 +59,7 @@ impl<const H: usize> TxInputNote<H> {
         note: &Note,
         merkle_opening: poseidon_merkle::Opening<(), H>,
         sk: &SecretKey,
-        skeleton_hash: BlsScalar,
+        payload_hash: &BlsScalar,
```

Comment:
Why this change? `BlsScalar`s are `Copy` as well.
```suggestion
        payload_hash: BlsScalar,
```